### PR TITLE
fix: tokens metadata is now resolved from more than one source/metadata standard

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Checklist

- [X] JIRA - [LW-10829](https://input-output.atlassian.net/browse/LW-10829)
- [x] Proper tests implemented
- [ ] Screenshots added.

---

1. Cardano makes no distinction between a fungible token (FT) and non-fungible token (NFT)
2. In Lace we refer to FTs as “tokens”
3. Metadata standards were developed to accommodate the needs of each, involving both on-chain and off-chain solutions, however issuers often mismatch since their utilising whatever is most convenient or aligned with their goals.
4. Lace originally implemented handling of tokens and NFTs to be based on the metadata attached, however this lead to unexpected results, despite us implementing a more pure approach.
5. Lace has since moved to detecting an NFT as simply an asset with a single supply
6. Tokens that are implementing the NFT standard are appearing in the correct section within Lace but the metadata is not being detected.

## Proposed solution

Token metadata now is selected based on a set of priorities, for FT, metadata is fetch from the token registry, if no metadata found, we will see if there is NFT metadata with FT attributes and get the metadata from that. 

[LW-10829]: https://input-output.atlassian.net/browse/LW-10829?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/browser-extension-wallet/src/utils/get-token-list.ts [hunk 1]
```diff
@@ -38,6 +44,123 @@ export interface GetTokenListParams {
...
+ * @param info - The asset information containing metadata for NFTs or tokens.
+ * @param environmentName - The name of the blockchain environment, used in default naming.
+ *
+ * @returns An object containing the name and image URL for the NFT.
+ */
+export const getNftDisplayMetadata = (
+  info: AssetOrHandleInfo,
+  environmentName: Wallet.ChainName
+): Pick<NFT, 'name' | 'image'> => {
+  const name = info.nftMetadata?.name || info.tokenMetadata?.name || `SingleNFT${environmentName || ''}`;
```
Comment (mkazlauskas) [inline apps/browser-extension-wallet/src/utils/get-token-list.ts:158]: `SingleNFTMainnet` is gibberish
```suggestion
  const name = info.nftMetadata?.name || info.tokenMetadata?.name || tryEncodeToUtf8(info.assetName) || "Unnamed"`;
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/utils/get-token-list.ts (hunk 1 above)
Comment (AngelCastilloB) [reply apps/browser-extension-wallet/src/utils/get-token-list.ts:158]: @mkazlauskas this was in the original implementation, can/will change if everyone agrees tho

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6014/9773774112/index.html) for [e342260f](https://github.com/input-output-hk/lace/pull/1263/commits/e342260f4e48165824e73aac4884c1b0ea07587d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 32     | 1      | 0       | 0     | 33    | ❌     |

File: apps/browser-extension-wallet/src/utils/get-token-list.ts (hunk 1 above)
Comment (AngelCastilloB) [reply apps/browser-extension-wallet/src/utils/get-token-list.ts]: preemptively done here https://github.com/input-output-hk/lace/pull/1263/commits/41a4749f818e4fce65496b7a61ea36ac826d911d

File: apps/browser-extension-wallet/src/utils/get-token-list.ts [hunk 2]
```diff
@@ -38,15 +45,148 @@ export interface GetTokenListParams {
...
+
+/**
+ * Constructs display metadata for an asset, prioritizing token metadata,
+ * then NFT metadata, and finally using a fallback.
+ *
+ * @param info The asset or handle information containing various metadata.
+ *
+ * @returns An object with name, ticker, decimals and logo of the asset.
+ */
+export const getTokenDisplayMetadata = (
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/src/utils/get-token-list.ts:149]: Nice name for these functions 😎

**Review by rhyslbw — APPROVED** [review]
Great work @AngelCastilloB 💪

File: packages/core/src/ui/components/ImageWithFallback/ImageWithFallback.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React, { useState } from 'react';
+
+export interface ImageWithFallbackProps
+  extends React.DetailedHTMLProps<React.ImgHTMLAttributes<HTMLImageElement>, HTMLImageElement> {
+  defaultSrc: string;
```
Comment (mkazlauskas) [inline packages/core/src/ui/components/ImageWithFallback/ImageWithFallback.tsx:5]: Matching name of the component would make it more intuitive
```suggestion
  fallbackSrc: string;
```

Comment (AngelCastilloB) [reply packages/core/src/ui/components/ImageWithFallback/ImageWithFallback.tsx]: done here https://github.com/input-output-hk/lace/pull/1263/commits/bbc30a91464477195fc51e1564e9b11e6c0295ce

**Review by mkazlauskas — APPROVED** [review]
Nice work!

**Review by pczeglik-iohk — APPROVED** [review]
Nicely done @AngelCastilloB 💪

**Pull request merged**